airflow/providers/google/cloud/operators/cloud_run.py

```diff
@@ -353,3 +353,130 @@ def _wait_for_operation(self, operation: operation.Operation):
+        )
+
+        return Service.to_dict(service)
+
+
+class CloudRunDeleteServiceOperator(GoogleCloudBaseOperator):
+    """
+    Deletes a Service without executing it. Pushes the deleted service to xcom.
+
+    :param project_id: Required. The ID of the Google Cloud project that the service belongs to.
+    :param region: Required. The ID of the Google Cloud region that the service belongs to.
+    :param service_name: Required. The name of the service to create.
+    :param service: Required. The service descriptor containing the configuration of the service to submit.
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields = ("project_id", "region", "gcp_conn_id", "impersonation_chain", "service_name")
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        service_name: str,
+        service: dict | Service,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.service_name = service_name
+        self.service = service
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        # self._validate_inputs()
+
+    def _validate_inputs(self):
+        missing_fields = [k for k in ["project_id", "region", "service_name", "service"] if not getattr(self, k)]
+        if not self.project_id or not self.region or not self.service_name or self.service:
+            raise AirflowException(
+                f"Required parameters are missing: {missing_fields}. These parameters be passed either as "
+                "keyword parameter or as extra field in Airflow connection definition. Both are not set!"
+            )
+
+    def execute(self, context: Context):
+        hook: CloudRunServiceHook = CloudRunServiceHook(
+            gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain
+        )
+
+        service = hook.delete_service(
+            service_name=self.service_name,
+            service=self.service,
+            region=self.region,
+            project_id=self.project_id,
+        )
+
+        return Service.to_dict(service)
```

Comment:
I think all delete operators don't return anything after deletion. Can you please provide API documentation for this method? I think it returns nothing in case the deletion went well

Reply:
in [docs](https://cloud.google.com/python/docs/reference/run/latest/google.cloud.run_v2.services.services.ServicesClient#google_cloud_run_v2_services_services_ServicesClient_delete_service), delete_service returns `google.api_core.operation.Operation`. The result type for the operation will be [Service](https://cloud.google.com/python/docs/reference/run/latest/google.cloud.run_v2.types.Service)